File: module-core/src/main/java/maple/expectation/error/exception/auth/InvalidApiKeyException.java
package maple.expectation.error.exception.auth;

import maple.expectation.error.CommonErrorCode;
import maple.expectation.error.exception.base.ClientBaseException;
import maple.expectation.error.exception.marker.CircuitBreakerIgnoreMarker;

/** 유효하지 않은 API Key 예외 */
public class InvalidApiKeyException extends ClientBaseException
    implements CircuitBreakerIgnoreMarker {

  public InvalidApiKeyException() {
    super(CommonErrorCode.INVALID_API_KEY);
  }
}

File: module-common/src/main/kotlin/maple/expectation/error/exception/InvalidApiKeyException.kt
package maple.expectation.error.exception

import maple.expectation.error.CommonErrorCode
import maple.expectation.error.exception.base.ClientBaseException

class InvalidApiKeyException : ClientBaseException {

    /**
     * Default constructor with static message
     */
    constructor() : super(CommonErrorCode.INVALID_API_KEY)

    /**
     * Constructor with custom message
     */
    constructor(message: String) : super(
        CommonErrorCode.INVALID_API_KEY,
        message
    )
}
